[16.0][MIG] l10n_es_pos: Migration to 16.0

danielduqma opened the pull request:
https://github.com/OCA/l10n-spain/issues/2526

Commit:
[ADD] Add l10n_es_pos

sbidoul committed:
[UPD] prefix versions with 8.0

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

xparsox committed:
[MIG] l10n_es_pos

acysos committed:
[ADD] l10n_es_pos: posbox support and translations

chienandalu committed:
[MIG] l10n_es_pos: Migration to 11.0

jonerikceberio committed:
[FIX] l10n_es_pos: typo in oldname (#886)

jonerikceberio committed:
[IMP] l10n_es_pos: Handle the sequence properly (#902)

pedrobaeza committed:
[FIX] l10n_es_pos: Import uninstall_hook

chienandalu committed:
[IMP] l10n_es_pos: move simpl. inv. reference to standard field (#945)

This way we don't need to make glue modules over l10n_es_pos every time
we need to make use of that info for a generic purpose.

chienandalu committed:
[MIG] l10n_es_pos: Migration to 12.0

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (31 of 31 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_pos/es/

chienandalu committed:
[FIX] l10n_es_pos: XmlReceipt doesn't show customer

chienandalu committed:
[FIX] l10n_es_pos: prevent duplicate or missing orders

chienandalu committed:
[FIX] l10n_es_pos: errors due to context

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (32 of 32 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_pos/es/

CarlesAntoli committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (32 of 32 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_pos/ca/

CarlosRoca13 committed:
[IMP] l10n_es_pos: black, isort, prettier

CarlosRoca13 committed:
[MIG] l10n_es_pos: Migration to v13.0

JordiBForgeFlow committed:
rename category of l10n_es_pos, as 'Point of Sale' is not valid in 13.0

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (25 of 25 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_pos/es/

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (25 of 25 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_pos/ca/

joao-p-marques committed:
[IMP] l10n_es_pos: black, isort, prettier

joao-p-marques committed:
[MIG] l10n_es_pos: Migration to 14.0

TT26435

joao-p-marques committed:
[IMP] l10n_es_pos: Export ticket correctly

Co-authored-by: Enrique Martín Zabalza <enriquemartin@digital5.es>

oca-travis committed:
[UPD] Update l10n_es_pos.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_pos/

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (29 of 29 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_pos/ca/

rauferdeveloper committed:
[14.0][FIX] l10n_es_pos: Incorrect label texts and translations

oca-travis committed:
[UPD] Update l10n_es_pos.pot

OCA-git-bot committed:
l10n_es_pos 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_pos/

Commit:
[FIX] Refactoring use of l10n_es_unique_id

manuelregidor committed:
[MIG]l10n_es_pos: Migration to 15.0

Commit:
[IMP] l10n_es_pos:
- En el caso de tener red, el sistema para obtener la numeración de las facturas simplificadas hara una peticion al servidor.
- En caso contrario usaremos el uid generado por odoo.

Commit:
[UPD] Update l10n_es_pos.pot

OCA-git-bot committed:
[UPD] README.rst

chienandalu committed:
[FIX] l10n_es_pos: offline simplified number

danielduqma committed:
[MIG] l10n_es_pos: Migration to 16.0

rafaelbn commented:
Aquí ya debe de quedar definitivamente corregido este problema https://github.com/OCA/l10n-spain/pull/2582 para no tener que hacer cambios complicados en una misma versión que hace que 2 tipos de instancias sean incompatibles.

pedrobaeza commented:
/ocabot migration l10n_es_pos

ao-landoo committed:
[IMP] l10n_es_pos: Extract logic to functions to ease inheritance

flachica commented on l10n_es_pos/models/pos_order.py:
```diff
@@ -0,0 +1,70 @@
+            pos.l10n_es_simplified_invoice_sequence_id.next_by_id()
+        return super()._process_order(pos_order, draft, existing_order)
+
+    def _get_fields_for_order_line(self):
+        fields = super()._get_fields_for_order_line()
+
+        fields += ["l10n_es_unique_id"]
+        return fields
```
```suggestion
```
El campo `l10n_es_unique_id` va a nivel de `pos.order` y no de `pos.order.line`

danielduqma replied:
Gracias por la revisión! El método se añadió en 8cf467a para mejorar la compatibilidad con `pos_restaurant` en la versión 14.0, lo he sustituido por `_get_fields_for_draft_order` en un commit aparte para un eventual cherrry-pick en versiones previas.

manuelregidor committed:
[FIX]l10n_es_pos: Wrong customer info when reprinting receit

OCA-git-bot committed:
l10n_es_pos 15.0.1.1.0

danielduqma committed:
[FIX] l10n_es_pos: pos_restaurant compatibility

Fix callable method when retrieving fields for order, originally added
in 8cf467a.

ao-landoo committed:
[IMP] l10n_es_pos: Add is_simplified_config field for inheritance in other modules

suker committed:
[DEL] l10n_es_pos: deleting duplicated field

danielduqma commented:
Se ha rebaseado el desarrollo y se han aplicado los cambios aprobados en https://github.com/OCA/l10n-spain/pull/2812 para este módulo en la rama 14.0.

danielduqma committed:
[FIX] l10n_es_pos: duplicated POS names

POS config names can be repeated, for example when running
`point_of_sale` tests. Hence, created sequences would be duplicated, so
with this change this is checked and prevented adding a suffix.

danielduqma committed:
[FIX] l10n_es_pos: validation

Change made in 1baf50b avoids showing customer popup on orders above
limit (invoiced). Moving back logic to `validateOrder` fixes this
problem.

Commit:
[UPD] Update l10n_es_pos.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
l10n_es_pos 15.0.2.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-15.0/l10n-spain-15.0-l10n_es_pos
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-15-0/l10n-spain-15-0-l10n_es_pos/

chienandalu commented:
Ten en cuenta también algunos cambios que ha habido que adaptar en el FW que se ha hecho en #2965

danielduqma commented:
> Ten en cuenta también algunos cambios que ha habido que adaptar en el FW que se ha hecho en #2965

Hecho! He aplicado los cambios del 15.0 en el histórico de la rama.

flachica commented on l10n_es_pos/static/src/js/PaymentScreen.js:
```diff
@@ -0,0 +1,36 @@
+                        order.to_invoice = true;
+                    }
+                }
+                super.validateOrder(isForceValidate);
```
```suggestion
                await super.validateOrder(isForceValidate);
```

danielduqma replied:
Corregido, gracias por la revisión!

fcvalgar reviewed (approved):
Revisión funcional.

Perfecto el proceso de validación. Muy intuitivo y claro. 

https://www.awesomescreenshot.com/video/17092554?key=6af6d010c88d816f27488b2f43775b96

Gracias @danielduqma.

RodrigoBM commented:
@oca/local-spain-maintainers ¿Alguien puede mergear el PR?

rafaelbn reviewed (approved):
¡Gran trabajo a todos!

@omar7r tu que eres experto de POS 😄 y PSC?

@chienandalu de acuerdo?

@pedrobaeza te animas?

pedrobaeza commented:
Necesito la confirmación de alguien de PoS.

rafaelbn commented:
@pedrobaeza yo lo he probado y soy PSC de PoS, ¿No te sirvo? 😄

pedrobaeza commented:
Es una review técnica de JavaScript, Rafa.

rafaelbn commented:
Aquí tienes la review funcional: https://www.loom.com/share/c9dc6f97080d4d0fb4f188b782370cee

He revisado tanto funcionamiento como secuencias y desglose correcto de impuestos, como operativa para complir con la normativa.

zamberjo commented:
Hola, puedes incluir el FIX https://github.com/OCA/l10n-spain/pull/3066 (15.0) (14.0 https://github.com/OCA/l10n-spain/pull/3053)?

chienandalu reviewed (approved):
:+1:

pedrobaeza commented:
/ocabot merge nobump

RodrigoBM commented:
> Hola, puedes incluir el FIX #3066 (15.0) (14.0 #3053)?

No le ha dado a mi compañero a añadirlo antes del mergeo, ¿puedes crear un PR con el fix y lo revisamos?

zamberjo commented:
Si, estoy en ello
